packages/sdk/src/Actions/EventAction.ts

```diff
@@ -526,21 +529,85 @@ export class EventAction extends DeployableTarget<
    * @public
    * @async
    * @param {ActionStep} actionStep - The action step to validate. Can be a function of event step.
-   * @param {?TxParams & { chainId?: number }} [params] - Additional parameters for validation, including known events, logs, and chain ID.
+   * @param {TxParams & { transactionHash?: Hex, chainId?: number }} params - Additional parameters for validation, including, hash, known events, logs, and chain ID.
    * @returns {Promise<boolean>}
    */
   public async isActionStepValid(
     actionStep: ActionStep,
-    params?: TxParams & { chainId?: number },
+    params: TxParams & { hash?: Hex; chainId?: number },
   ) {
+    const client = this._config.getClient({
+      chainId: params.chainId,
```

Comment:
since this parameter is optional, should we default to our `__DEFAULT_CHAIN_ID__`?

Reply:
I've actually went ahead and moved the client to the point where `chainId` is required (but still optional in the params cause if you pass logs in, then you won't need `chainId`)

I think having to explicitly define the chain is more clear esp if you need to do anything on different chains